src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -36,12 +37,15 @@ abstract class BTypes {
    * name. The method assumes that every class type that appears in the bytecode exists in the map
    */
   def cachedClassBType(internalName: InternalName): Option[ClassBType] =
-    classBTypeCache.get(internalName)
+    Option(cachedClassBTypeOrNull(internalName))
+
+  def cachedClassBTypeOrNull(internalName: InternalName): ClassBType =
+    classBTypeCache.map.get(internalName)
 
   // Concurrent maps because stack map frames are computed when in the class writer, which
   // might run on multiple classes concurrently.
   // Note usage should be private to this file, except for tests
-  val classBTypeCache: concurrent.Map[InternalName, ClassBType] = recordPerRunCache(FlatConcurrentHashMap.empty)
+  val classBTypeCache: ClearableJConcurrentHashMap[InternalName, ClassBType] = recordPerRunCache(new ClearableJConcurrentHashMap[InternalName, ClassBType])
```

Comment:
Could you use `frontendAccess.recordPerRunJavaMapCache` instead?

Reply:
I tried, but it will take some rework of `unrecordCache` / `keepPerRunCachesAfterRun` that eluded me.

Reply:
ah :(